test/e2e/usersignup_test.go

```diff
@@ -45,662 +43,190 @@ func (s *userSignupIntegrationTest) TearDownTest() {
-	require.NoError(s.T(), err)
-	s.T().Logf("UserSignup '%s' created", userSignup.Name)
-
-	// Confirm the UserSignup was created
-	_, err = s.hostAwait.WaitForUserSignup(userSignup.Name)
-	require.NoError(s.T(), err)
-
-	// Confirm that:
-	// 1) the Complete condition is set to true
-	_, err = s.hostAwait.WaitForUserSignup(userSignup.Name, wait.UntilUserSignupHasConditions(
-		v1alpha1.Condition{
-			Type:   v1alpha1.UserSignupApproved,
-			Status: corev1.ConditionTrue,
-			Reason: "ApprovedAutomatically",
-		},
-		v1alpha1.Condition{
-			Type:   v1alpha1.UserSignupComplete,
-			Status: corev1.ConditionTrue,
-		}))
-	require.NoError(s.T(), err)
-}
-
-func (s *userSignupIntegrationTest) TestUserSignupWithAutoApprovalWhenMultipleMURAlreadyExists() {
-	// Set the user approval policy to automatic
-	s.setApprovalPolicyConfig("automatic")
-
-	// Create a MUR
-	s.T().Logf("Creating MasterUserRecord with namespace %s", s.namespace)
-	userID := uuid.NewV4().String()
-	mur := s.newMasterUserRecord("robert-at-hotel-com", userID)
-	err := s.awaitility.Client.Create(context.TODO(), mur, testsupport.CleanupOptions(s.testCtx))
-	require.NoError(s.T(), err)
-	s.T().Logf("MasterUserRecord '%s' with label %s created", mur.Name, mur.Labels["toolchain.dev.openshift.com/user-id"])
-
-	// Confirm the MasterUserRecord was created
-	_, err = s.hostAwait.WaitForMasterUserRecord(mur.Name)
-	require.NoError(s.T(), err)
-
-	// Create a second MUR with the same UserID but different name
-	s.T().Logf("Creating MasterUserRecord with namespace %s", s.namespace)
-	mur = s.newMasterUserRecord("another-robert-at-hotel-com", userID)
-	err = s.awaitility.Client.Create(context.TODO(), mur, testsupport.CleanupOptions(s.testCtx))
-	require.NoError(s.T(), err)
-	s.T().Logf("MasterUserRecord '%s' with label %s created", mur.Name, mur.Labels["toolchain.dev.openshift.com/user-id"])
-
-	// Confirm the MasterUserRecord was created
-	_, err = s.hostAwait.WaitForMasterUserRecord(mur.Name)
-	require.NoError(s.T(), err)
-
-	// Create user signup with the same UserID as the MURs
-	s.T().Logf("Creating UserSignup with namespace %s", s.namespace)
-	userSignup, err := newUserSignup(s.awaitility.Host(), userID, "robert@hotel.com")
-	require.NoError(s.T(), err)
-
-	err = s.awaitility.Client.Create(context.TODO(), userSignup, testsupport.CleanupOptions(s.testCtx))
-	require.NoError(s.T(), err)
-	s.T().Logf("UserSignup '%s' created", userSignup.Name)
-
-	// Confirm the UserSignup was created
-	_, err = s.hostAwait.WaitForUserSignup(userSignup.Name, wait.UntilUserSignupHasConditions(
-		v1alpha1.Condition{
-			Type:    v1alpha1.UserSignupComplete,
-			Status:  corev1.ConditionFalse,
-			Reason:  "InvalidMURState",
-			Message: "multiple matching MasterUserRecord resources found",
-		}))
-	require.NoError(s.T(), err)
-}
+func (s *userSignupIntegrationTest) checkUserSignupManualApproval() {
+	s.T().Run("usersignup created first then manually approved", func(t *testing.T) {
```

Comment:
I guess, that the same problem with the parent `t` being nill we can face here as well.